packages/grid/src/vaadin-grid-selection-column.js

```diff
@@ -149,6 +162,132 @@ class GridSelectionColumn extends GridColumn {
+        const dy = scrollableArea.top + scrollTriggerArea - this.__lassoCurrentY;
+        const percentage = Math.min(1, dy / scrollTriggerArea);
+        this._grid.$.table.scrollTop -= percentage * maxScrollAmount;
+      }
+      if (this.__lassoDy > 0 && this.__lassoCurrentY > scrollableArea.bottom - scrollTriggerArea) {
+        const dy = this.__lassoCurrentY - (scrollableArea.bottom - scrollTriggerArea);
+        const percentage = Math.min(1, dy / scrollTriggerArea);
+        this._grid.$.table.scrollTop += percentage * maxScrollAmount;
+      }
+
+      // Schedule the next auto scroll
+      setTimeout(() => this.__lassoAutoScroller(), 10);
+    }
+  }
+
+  /**
+   * Gets the scrollable area of the grid as a bounding client rect. The
+   * scrollable area is the bounding rect of the grid minus the header and
+   * footer.
+   *
+   * @private
+   */
+  __getScrollableArea() {
+    const gridRect = this._grid.$.table.getBoundingClientRect();
+    const headerRect = this._grid.$.header.getBoundingClientRect();
+    const footerRect = this._grid.$.footer.getBoundingClientRect();
+
+    return {
+      top: gridRect.top + headerRect.height,
+      bottom: gridRect.bottom - footerRect.height,
+      left: gridRect.left,
+      right: gridRect.right,
+      height: gridRect.height - headerRect.height - footerRect.height,
+      width: gridRect.width,
+    };
+  }
+
+  /** @private */
+  __onSelectionColumnCellTrack(event) {
+    if (!this.selectRowsByDragging) {
+      return;
+    }
+    this.__lassoCurrentY = event.detail.y;
+    this.__lassoDy = event.detail.dy;
+    if (event.detail.state === 'start') {
+      this._grid.setAttribute('disable-text-selection', true);
+      // ensure text selection is cleared in Firefox as user-select: none; does not work properly
+      this.__clearTextSelection();
+      this.__lassoWasEnabled = true;
+      const renderedRows = this._grid._getRenderedRows();
+      // Get the row where the drag started
+      const lassoStartRow = renderedRows.find((row) => row.contains(event.currentTarget.assignedSlot));
+      // Whether to select or deselect the items on drag
+      this.__lassoSelect = !this._grid._isSelected(lassoStartRow._item);
+      // Store the index of the row where the drag started
+      this.__lassoDragStartIndex = lassoStartRow.index;
+      // Start the auto scroller
+      this.__lassoAutoScroller();
+    } else if (event.detail.state === 'end') {
+      this.__lassoDragStartIndex = undefined;
+      this._grid.removeAttribute('disable-text-selection');
+    }
+  }
+
+  /** @private */
+  __clearTextSelection() {
+    if (window.getSelection) {
+      if (window.getSelection().empty) {
+        // Chrome
+        window.getSelection().empty();
+      } else if (window.getSelection().removeAllRanges) {
+        // Firefox
+        window.getSelection().removeAllRanges();
+      }
+    } else if (document.selection) {
+      // IE?
+      document.selection.empty();
```

Comment:
IE is not supported

Reply:
This is going to be removed in favor of preventing default on mousedown event.